internal/tofu/node_resource_abstract_instance.go

```diff
@@ -348,6 +348,32 @@ func (n *NodeAbstractResourceInstance) writeResourceInstanceStateImpl(ctx EvalCo
 	return nil
 }
 
+// planForget returns a removed from state diff.
+func (n *NodeAbstractResourceInstance) planForget(ctx EvalContext, currentState *states.ResourceInstanceObject, deposedKey states.DeposedKey) (*plans.ResourceInstanceChange, tfdiags.Diagnostics) {
```

Comment:
Why does this return diags?  I don't see anywhere that they are used?  Is this following the prototype of another similar function?

Reply:
They are actually used [here](https://github.com/opentofu/opentofu/pull/1158/files/f0e276135196719cfde459ef80bdf64346d0d1ba#diff-846b5b253854eec7652df68e77a9ed83ff5346529c138439120a9998a2bf98fdR152) and [here](https://github.com/opentofu/opentofu/pull/1158/files/f0e276135196719cfde459ef80bdf64346d0d1ba#diff-f305d1d715be433c5e7d826409091c8f34b27dfebf25d488b3fadcb6145c640aR155) 🙃

Reply:
Thanks!

Reply:
Sorry, I should have specified that diags is never set within this function and always returns null